src/implementations/vanilla/src/basics/form-elements/text-field/text-field.html

```diff
@@ -11,6 +11,7 @@
             <!--SLOT-->
             <input name="{{name}}" id="{{id}}" class="hig__text-field__input hig__text-field__input--no-value" placeholder="{{placeholder}}"
                 value="{{value}}">
+            <button class="hig__text-field__clear-button"></button>
```

Comment:
I have a feeling this will break the appearance of the label.
The label gets some of it's appearance from the state of the input using the adjacent sibling selector.
Try moving the clear button after the label.